src/App.tsx

```diff
@@ -1,24 +1,142 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { FilterTodos } from './components/FilterTodos';
+import { Form } from './components/Form';
+import { NotificationError } from './components/NotificationError';
 
-const USER_ID = 0;
+import { FilterType } from './types/FilterType';
+import { ErrorType } from './types/ErrorType';
+
+const USER_ID = 10210;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [error, setError] = useState<ErrorType | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isProcessingId, setIsProcessingId] = useState<number[]>([]);
+
+  const getTodosFromServer = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setError(ErrorType.LOAD);
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+  };
+
+  const postTodo = useCallback(async (title: string) => {
+    try {
+      const newTodo = {
+        userId: USER_ID, completed: false, title,
+      };
+      const temp = { ...newTodo, id: 0 };
+
+      setLoading(true);
+      setTempTodo(temp);
+      await addTodo(newTodo);
+      await getTodosFromServer();
+    } catch {
+      setError(ErrorType.ADD);
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
```

Comment:
You should clear all timers on the component unmount
https://bobbyhadz.com/blog/react-cleartimeout